### PR TITLE
add various types to support velero/snapshots in EC

### PR DESCRIPTION
**Pull request by laverya**

### CODE REVIEW

File: apis/v1beta1/config_types.go [hunk 1]
```diff
@@ -62,8 +62,18 @@ type Roles struct {
 	Custom     []NodeRole `json:"custom,omitempty"`
 }
 
+type VeleroBuiltin struct {
```
Comment (ricardomaraschini) [inline apis/v1beta1/config_types.go:65]: Can we have some comments on these new structs explaining why they exist ? Maybe showing up an use case.

File: types/release_metadata.go [hunk 2]
```diff
@@ -5,11 +5,12 @@ import "github.com/k0sproject/k0s/pkg/apis/k0s/v1beta1"
...
-	Configs       v1beta1.HelmExtensions
-	AirgapConfigs v1beta1.HelmExtensions
-	Protected     map[string][]string
+	Versions       map[string]string
+	K0sSHA         string
+	K0sBinaryURL   string
+	K0sImages      []string
+	Configs        v1beta1.HelmExtensions            // always applied
+	AirgapConfigs  v1beta1.HelmExtensions            // applied for airgap installations
+	BuiltinConfigs map[string]v1beta1.HelmExtensions // applied if the relevant builtin addon is enabled
```
Comment (ricardomaraschini) [inline types/release_metadata.go:14]: Is this field intended for usage with other builtin extensions as well (other than Velero) ? Let's suppose I want to override OpenEBS config, will I be able to use this field for it ?

Comment (laverya) [reply types/release_metadata.go:14]: This is intended for any future things we add as 'builtin' items, like distributed storage when that becomes a thing

File: apis/v1beta1/config_types.go (hunk 1 above)
Comment (laverya) [reply apis/v1beta1/config_types.go:65]: sure

**Review by ricardomaraschini — APPROVED** [review]

**Pull request merged**